File: CPP/Komputer.cpp
#include "Cpu.cpp"
#include "Ram.cpp"
#include "Harddrive.cpp"

class Komputer {
private:
    string nama;
    Cpu cpu;
    Ram ram[10];
    Harddrive harddrive;
    int jumlahRam;
public:
    Komputer() { jumlahRam = 0; }
    void setNama(string nama) { this->nama = nama; }
    void setCpu(Cpu cpu) { this->cpu = cpu; }
    void setHarddrive(Harddrive harddrive) { this->harddrive = harddrive; }
    void tambahRam(Ram r) {
        if (jumlahRam < 10) {
            ram[jumlahRam++] = r;
        } else {
            cout << "RAM penuh!\n";
        }
    }
    void tampil() {
        cout << "Informasi Komputer:" << endl;
        cout << "Nama      : " << nama << endl;
        cpu.tampil();
        for (int i = 0; i < jumlahRam; i++) {
            ram[i].tampil();
        }
        harddrive.tampil();
    }
};

File: Python/Komputer.py
from Cpu import Cpu
from Ram import Ram
from Harddrive import Harddrive

class Komputer:
    def __init__(self, nama):
        self.nama = nama
        self.cpu = None
        self.ram_list = []
        self.harddrive = None
    
    def set_cpu(self, cpu):
        self.cpu = cpu
    
    def tambah_ram(self, ram):
        self.ram_list.append(ram)
    
    def set_harddrive(self, harddrive):
        self.harddrive = harddrive
    
    def tampil(self):
        print(f"Nama      : {self.nama}")
        if self.cpu:
            print(f"Cpu       : {self.cpu.get_info()}")
        for ram in self.ram_list:
            print(f"Ram       : {ram.get_info()}")
        if self.harddrive:
            print(f"Harddrive : {self.harddrive.get_info()}")